src/EnergyPlus/DataPlant.hh

```diff
@@ -597,6 +601,7 @@ namespace DataPlant {
 		int IndexInLoopSidePumps; // If I'm a pump, this tells my index in PL(:)%LS(:)%Pumps
 		Real64 TempDesCondIn;
 		Real64 TempDesEvapOut;
+		std::shared_ptr< PlantComponent > compPtr;
```

Comment:
Yeah, I don't recall asking @kbenne for his thoughts...  

So I am trying to refactor in such a way that doesn't disrupt the existing framework (so that we can run to verify no-diffs at all stops), and so that the team is able to gradually move toward the refactored approach instead of a dead stop do-over.  Thus, I leave the objects to live in the namespace in a simple vector.  I tried leaving them as plain objects living in the namespace, not pointers at all, but well that went bad.  The higher level plant data structures need a pointer to the instance that lives at a particular location in the loop/branch/comp topology.  Since, for example, a chiller, can live at up to 3 locations, there are 3 variables in the loop/branch/comp topology that need to all point at the same instance of the chiller.  Thus it became a shared pointer.